### DISCUSSION TITLE
skywalking could not get the trace info between sonarqube(7.6) and mysql(5.7) which works well between sonarqube(8.9) and postsql(13.8)

### DISCUSSION THREAD

**Discussion by arielwei123** [apache/skywalking#10891] [Q&A] [answered]
I met a issue, that my skywalking could not get the trace info between sonarqube(7.6) and mysql(5.7) , and mysql service could not be found under virtual database page. If I changed to use sonarqube(8.9) and postsql(13.8), the postsql service works well under  virtual database page.
    The k8s deploy file for sonarqueb+mysql shown below.
    Very appreciate if give some suggestions, thanks!

         env:
         - name: SONARQUBE_JDBC_USERNAME
           value: sonar            
         - name: SONARQUBE_JDBC_PASSWORD
           value: sonar
         - name: SONARQUBE_JDBC_URL
           value: "jdbc:mysql://mysql-agent:3306/sonardb?useSSL=false&useUnicode=true&characterEncoding=utf8&rewriteBatchedStatements=true&useConfigs=maxPerformance&serverTimezone=UTC"
         - name: SONAR_WEB_JAVAADDITIONALOPTS
           value: "-javaagent:/skywalking/agent/skywalking-agent.jar=agent.service_name=sonar-mysql-agent=web"  
           ### following for agent
         - name: LOGGING_LEVEL_ROOT
           value: "DEBUG"
         - name: SW_LOGGING_OUTPUT
           value: "FILE"
         - name: SW_LOGGING_DIR
           value: "/skywalking/"
         - name: JAVA_TOOL_OPTIONS
           value: "-javaagent:/skywalking/agent/skywalking-agent.jar"
         - name: SW_AGENT_NAME
           value: "sonar-mysql-agent"
         - name: SW_AGENT_COLLECTOR_BACKEND_SERVICES
           value: "skywalking-oap:11800"
         - name: SW_AGENT_INSTANCE_NAME
           value: sonar-mysql-agent

The mysql-agent:3306 should also be seen in the following page, 
![图片](https://github.com/apache/skywalking/assets/135794581/2245bfee-59d7-4619-9636-202ef0862ed8)

**Comment by arielwei123** [comment]
By the way, the skywalking-oap verion I use is 9.2.

**Comment by wu-sheng** [comment]
I don't think we have sonarqube plugin at all, do we?
I think the root is, which JDBC is being used.

**Comment by arielwei123** [comment]
Sonarqube support both JDBC with mysql or postsql, the only different is before sonar7.8 mysql supported , after sonar7.8  mysql not supported.

**Comment by arielwei123** [comment]
Do we need any special plugin for sonarque in skywalking?

**Reply by wu-sheng** [reply]
We don't know that.

**Reply by wu-sheng** [reply]
This is like you are asking an answer from ppl never faced it. The worlds have thousands of frameworks actively running, we don't know many of them. It is only up to you to decide which is required.

**Comment by arielwei123** [comment]
Yes, I see.  I actually want to know if I miss any necessary plugins or deploy parameters when use skywalking trace mysql, since I just start to learn how to use it. Maybe I could change to try another application(not sonar) with mysql.  Any suggestion for the demo configuration for tracing mysql?

**Reply by arielwei123** [reply, marked as answer]
My issue resolved by mount sonar.properties file out of  container  to set Java env values shown below. It's done. Seems the Java env not take effect when just set in .yaml file. 
sonar.web.javaOpts=-Xmx1024m -Xms128m -XX:+HeapDumpOnOutOfMemoryError -javaagent:/skywalking/agent/skywalking-agent.jar
sonar.web.javaAdditionalOpts=-javaagent:/skywalking/agent/skywalking-agent.jar
